server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -735,7 +735,9 @@ protected void doRun() {
                         originalThread
                     );
 
-                    i++;
+                    validateIndexRequestPostProcessing(requestSlot, indexRequest, onFailure);
```

Comment:
`executePipelines` is (well, can be) asynchronous, you can't call  `validateIndexRequestPostProcessing` here like this.